hi-ml-azure/testazure/testazure/test_himl.py

```diff
@@ -119,6 +120,103 @@ def test_write_run_recovery_file(mock_run: mock.MagicMock) -> None:
     assert expected_run_recovery_id != recovery_file_text
 
 
+@pytest.fixture(scope="module")
+def dummy_max_num_nodes_available() -> int:
+    return randint(2, 10)
+
+
+@pytest.fixture(scope="module")
+def dummy_compute_cluster_name() -> str:
+    return 'dummy_cluster'
+
+
+@pytest.fixture(scope="module")
+def mock_scale_settings(dummy_max_num_nodes_available: int) -> MagicMock:
+    return MagicMock(maximum_node_count=dummy_max_num_nodes_available)
+
+
+@pytest.fixture(scope="module")
+def mock_compute_cluster(mock_scale_settings: MagicMock) -> MagicMock:
+    return MagicMock(scale_settings=mock_scale_settings)
+
+
+@pytest.fixture(scope="module")
+def mock_workspace(mock_compute_cluster: MagicMock, dummy_compute_cluster_name: str) -> MagicMock:
+    return MagicMock(compute_targets={dummy_compute_cluster_name: mock_compute_cluster})
+
+
+@pytest.mark.fast
+def test_validate_num_nodes(dummy_max_num_nodes_available: int, dummy_compute_cluster_name: str,
+                            mock_workspace: MagicMock) -> None:
+    # If number of requested nodes <= max available nodes, nothing should happen
+    num_nodes_requested = dummy_max_num_nodes_available // 2
+    existing_compute_clusters = mock_workspace.compute_targets
+    print(f"Existing compute cluster scale settings: {existing_compute_clusters['dummy_cluster'].scale_settings}")
+    himl.validate_num_nodes(existing_compute_clusters, dummy_compute_cluster_name, num_nodes_requested)
+    num_nodes_requested = dummy_max_num_nodes_available
+    himl.validate_num_nodes(existing_compute_clusters, dummy_compute_cluster_name, num_nodes_requested)
+
+    # But if number of nodes requested > max available, a ValueError should be raised
+    num_nodes_requested = randint(dummy_max_num_nodes_available, 5000)
+    with pytest.raises(ValueError) as e:
+        himl.validate_num_nodes(existing_compute_clusters, dummy_compute_cluster_name, num_nodes_requested)
+    expected_error_msg = f"You have requested {num_nodes_requested} nodes, which is more than your compute cluster's"
+    f"maximum of {dummy_max_num_nodes_available} nodes"
+    assert expected_error_msg in str(e)
```

Comment:
Have you considered using the `match` kwarg of `pytest.raises` instead?

Reply:
nice one, I did not know about that!

Reply:
Neither did I! I found it googling around. I wasn't very comfortable hard-coding messages from somewhere else in the codebase and tried to find an alternative method, but I guess there's no other way to do it.

Reply:
I don't understand how this helps? We still have to provide the message, but with match we specify it inside the call to `raises`, rather than outside?

Reply:
I think `match` is generally more powerful as it uses regex (it calls `re.search`). But it's not critical here.

Reply:
I see, thanks. Happy to update (for the chance to try out a feature I didn't know about!)